### ISSUE TITLE
Facility locator location and service type fields do not announce dropdown status after user types in field.

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22830]
## Open questions

---
**Note: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22886 and https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22926 should be completed before working on this ticket.**

## Describe the defect
When typing into the location field and service type fields, the dropdown opens, but does not reliably announce to screenreaders that it is now open. 

- For the VA Health > Service type field, aria-expanded is getting updated correctly, but no announcement is heard. 
  - If you type only a single character, then pause, into the service type input you can hear a partial 'expanded' announcement before characters typed is repeated in VO
  - If you type multiple characters, no announcement is heard. This is _likely_ due to the announcement of the typed characters in the input 'overriding' the announcement of the dropdown status.
- The Community providers > Servicetype field is not updating aria-expanded correctly or announcing the dropdown status reliably. 
- Location field is not announcing the dropdown status at all
  - Also not getting aria-expanded set correctly. 

Additionally, neither location or service type are announcing to screenreader users when the options in the dropdown change. Adding a 'results summary' style status update as the user types will let the user know that results are available and how many results their query has narrowed the list to. 

## To Reproduce
1. Go to [staging facility locator prototype](https://staging.va.gov/find-locations/)
2. Turn on your screenreader of choice
3. Click or tab into location field and type a few characters
4. See the dropdown opens with filtered options. 
5. But note that there is no announcement of that dropdown opening. 
6. Type another character or delete to change the list of options. 
7. Note that there is no announcement of the updated content. 
8. Repeat in the VA health > service and Community provider > Service type type fields.

To see the dropdown being announced when you click (this is expected behavior):
1. Click into location or service type field
2. See the dropdown open or close
3. Hear the screenreader announce if the dropdown is expanded or collapsed  

## Screenshots
- Video showing [single character typed into the VA Health > Service Type input](https://github.com/user-attachments/assets/6b90b2db-e400-4294-a295-f50f6d28c056). You can hear the screenreader announce the dropdown is expanded. 
- Video showing [multiple characters typed in sequence into the VA Health > Service Type input](https://github.com/user-attachments/assets/d8836d42-4675-4244-a848-373494c36097). You do not hear the dropdown announcement at all. 

## Recommendation
Adding a results summary field will alert screenreader users when the dropdown has options for them to explore. Updating this status as the user continues to type keeps the user informed as the options in the dropdown update.

-  Add a 'results summary' div to the component, similar to [VADS combobox](https://design.va.gov/components/form/combo-box)
- Give it [role="status"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/status_role) and ensure it is present on the page before it has content.
  - The VADS status message also adds these classes, `.usa-combo-box__status` and `.usa-sr-only`, use them if possible. Otherwise, we'll need to visually hide the element in some other fashion.
- Add the text content dynamically when the dropdown opens. "x results available."
- Update the content as the user continues typing or deleting to reflect the number of results.
- If no results are available, the text content should read "No results available."
- When dropdown closes, clear the text content.

## Additional context
- Default dowshift announces the number of results and instructions for navigating dropdown.
<img width="476" height="339" alt="Image" src="https://github.com/user-attachments/assets/4d3a25a5-a249-4047-b193-4d46f1e2c451" />
 
- [VADS combobobx](https://design.va.gov/components/form/combo-box#examples) uses a 'results summary' that reads 'xx results available.' The dropdown opening is not announced reliably when typing into the VADS combobox. Similar to ours, when a single character is typed, you can hear the 'expanded' announcement, but when multiple characters are typed, you do not hear the dropdown state announced.
<img width="1231" height="324" alt="Image" src="https://github.com/user-attachments/assets/b19dcdf6-4c17-46de-93c3-3a04ca879c53" />

- [VADS search with typeahead](https://design.va.gov/components/search-input#with-typeahead) announces expanded/collapsed when a user types, but does not offer any results summary type info.

## AC / Expected behavior
- [ ] When a user types into the location field and the dropdown opens, their screenreader announces the dropdown has options.
- [ ] As a user continues to type into the location field and the dropdown options change, their screenreader announces the updated options.
 
- [ ] When a user types into the VA Health > service type field and the dropdown opens, their screenreader announces the dropdown has options.
- [ ] As a user continues to type into the VA Health > service type field and the dropdown options change, their screenreader announces the updated options.

- [ ] When a user types into the Community providers > service type field and the dropdown opens, their screenreader announces the dropdown has options.
- [ ] As a user continues to type into the Community providers > service type field and the dropdown options change, their screenreader announces the updated options.

## Definition of done
- [ ] Code completed; PR created
- [ ] Review instance created/available; internal team notified to start reviews
- [ ] Code review (meets established quality and security standards)
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [ ] Functional/feature review (meets acceptance criteria)
- [ ] Automated tests passed; PR approved
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by davidmpickett** [comment]
> [@qjac](https://github.com/qjac) / [@davidmpickett](https://github.com/davidmpickett) Also additional caveat for service type field, it does not currently announce the dropdown and available options when on focus before the user starts typing, indicating that user can use the field for typing/searching AND as a dropdown.

Can you add reproduction steps and/or video? I'm not understanding the scenario.

**Comment by r-bartlett-gsa** [comment]
> > [@qjac](https://github.com/qjac) / [@davidmpickett](https://github.com/davidmpickett) Also additional caveat for service type field, it does not currently announce the dropdown and available options when on focus before the user starts typing, indicating that user can use the field for typing/searching AND as a dropdown.
> 
> Can you add reproduction steps and/or video? I'm not understanding the scenario.

@davidmpickett I observed that in the video you shared https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830#issuecomment-3603482682 It's announcing "Begin typing to search for service..." however, there's no announcement that the entire dropdown list can be navigated without typing anything. Essentially, the service type field can be used two ways, as a dropdown and as an autosuggest. Visually, most users could tell that there's a dropdown (although some missed it during testing), but for AT users there are no indication that it's a dropdown. Is that needed?

**Comment by davidmpickett** [comment]
> Announcing the number of suggestions might be useful for AT users as that provides context and scope of available options to navigate. It also tells the users that the suggestions list is dynamic and changing as they type.

I agree with this. The VA Combo box handles this really elegantly as you type or delete letters:

https://github.com/user-attachments/assets/524c65ec-434a-49b6-af77-1a52665982ec

If it's feasible to match this behavior, I would be in favor of that.

**Comment by carlyMhayter** [comment]
Hi @r-bartlett-gsa! I just wanted to check that this work is not blocked by Bryan's 3rd PR as formerly noted? Should I wait until his work is merged?

**Comment by r-bartlett-gsa** [comment]
@qjac Can you please verify these updates on this RI: https://0392e52e96da7f790d8c547adc15c7b8.vfs.va.gov/

**Comment by qjac** [comment]
Tested in Edge using JAWS. Could not test the Community provider > Service type field on the RI, but the VA Health > Service type and Location fields are announcing the number of results or no results consistently. ✅
